loopchain/peer/block_manager.py

```diff
@@ -578,50 +577,33 @@ def __block_request_to_peers_in_sync(self, peer_stubs, my_height, unconfirmed_bl
                         unconfirmed_block_height = current_unconfirmed_block_height
 
                 try:
-                    result = True
-                    if max_height == unconfirmed_block_height == block.header.height \
-                            and max_height > 0 and not confirm_info:
+                    if (max_height == unconfirmed_block_height == block.header.height and
+                            max_height > 0 and not confirm_info):
                         self.candidate_blocks.add_block(
                             block, self.blockchain.find_preps_addresses_by_header(block.header))
                         self.blockchain.last_unconfirmed_block = block
-                        result = True
                     else:
-                        result = self.__add_block_by_sync(block, confirm_info)
+                        self.__add_block_by_sync(block, confirm_info)
 
-                    if result:
-                        if block.header.height == 0:
-                            self.__rebuild_nid(block)
-                        elif self.blockchain.find_nid() is None:
-                            genesis_block = self.blockchain.find_block_by_height(0)
-                            self.__rebuild_nid(genesis_block)
+                    if block.header.height == 0:
+                        self.__rebuild_nid(block)
+                    elif self.blockchain.find_nid() is None:
+                        genesis_block = self.blockchain.find_block_by_height(0)
+                        self.__rebuild_nid(genesis_block)
 
                 except KeyError as e:
-                    result = False
-                    util.logger.error("fail block height sync: " + str(e))
-                    break
+                    util.logger.error(f"fail block height sync: {type(e), e}")
+                    raise
```

Comment:
Do you know when this exception could be happened?

Reply:
and.. I think more specific logging message needed. It looks very similar to log msg `except Exception as e` below

Reply:
No idea for what triggers KeyError, but I've added traceback.